### DISCUSSION TITLE
Multiple non-Maven jars

### DISCUSSION THREAD

**Discussion by Sailsman63** [com-lihaoyi/mill#1769] [Q&A] [answered]
I've been trying to add multiple non-Maven jars to the `unmanagedClasspath`, but I'm obviously misunderstanding something.

```
 def unmanagedClasspath = Agg(
     mill.modules.Util.download(
       "https://github.com/blackears/svgSalamander/releases/download/v1.1.3/svgSalamander-1.1.3.jar",
       "svgSalamander.jar"
     )
    , mill.modules.Util.download(
      "https://github.com/beanshell/beanshell/releases/download/2.1.0/bsh-2.1.0.jar",
      "bsh.jar"
    )
 )
```

When I try to compile the script (just `mill resolve __.`) I get the following:

```
/mnt/develop/Shared/Source/ArtOfIllusion/build.sc:16: type mismatch;
 found   : String("svgSalamander.jar")
 required: os.RelPath
       "svgSalamander.jar"
       ^
/mnt/develop/Shared/Source/ArtOfIllusion/build.sc:20: type mismatch;
 found   : String("bsh.jar")
 required: os.RelPath
      "bsh.jar"
      ^
/mnt/develop/Shared/Source/ArtOfIllusion/build.sc:13: `T.command` definitions must have 1 parameter list
 def unmanagedClasspath = Agg(
```

The first two errors I really don't get, as the manual entry for downloading non-Maven jars clearly shows a string in that position. Is the manual outdated? The third error I have no clue about at all.

Using mill 0.10.0

**Comment by lefou** [comment, marked as answer]
You found an error in the documentation. `Util.download` requires a `os.RelPath` as second parameter, not a `String`. The fix is rather straight forward, you can create a `RelPath` with `os.rel`:

```scala
 def unmanagedClasspath = Agg(
     mill.modules.Util.download(
       "https://github.com/blackears/svgSalamander/releases/download/v1.1.3/svgSalamander-1.1.3.jar",
       os.rel / "svgSalamander.jar"
     )
    , mill.modules.Util.download(
      "https://github.com/beanshell/beanshell/releases/download/2.1.0/bsh-2.1.0.jar",
      os.rel / "bsh.jar"
    )
 )
```

The thirds error is an aftereffect of the first two. Because the body of the target definition doesn't type check, the compiler can't apply all the other things is usually does, e.g. conversions and macros.

**Reply by lefou** [reply]
* FTR, https://github.com/com-lihaoyi/mill/pull/1770

**Reply by Sailsman63** [reply]
Thanks! That moves me forward.
